pyqtgraph/graphicsItems/ViewBox/ViewBox.py

```diff
@@ -87,7 +87,7 @@ class ViewBox(GraphicsWidget):
     sigYRangeChanged = QtCore.Signal(object, object)
     sigXRangeChanged = QtCore.Signal(object, object)
     sigRangeChangedManually = QtCore.Signal(object)
-    sigRangeChanged = QtCore.Signal(object, object)
+    sigRangeChanged = QtCore.Signal(object, object, object)
```

Comment:
this change may break a lot of code out there... might be worth seeing if we can do something with `sigYRangeChanged` and `sigXRangeChanged` independently.  Not saying making this change is out of the question; just pointing out that we should at least examine if we can do without it.

Reply:
I agree that this is problematic, but it won't break as much code as you think right now:
https://doc.qt.io/qtforpython/overviews/signalsandslots.html
"The signals and slots mechanism is type safe: The signature of a signal must match the signature of the receiving slot. (In fact a slot may have a shorter signature than the signal it receives because it can ignore extra arguments.)"

I came across that behavior because we are doing that already:
The previous definition of `viewRangeChanged` (which this signal get connected to) was
`def viewRangeChanged(self)`
So this has been silently discarding the (two) objects already.
The third, 'changed' object should likewise be discarded by other code, although I am unsure of the behavior if the other code uses an explicit `@Slot()` decorator.

Working with the sigXRangeChanged and sigYRangeChanged signals individually is difficult, because ideally, a range change might be simultaneously announced for both x and y axis. That should be handled by single data update.
If we react to the individual signals, we will receive two notifications, with no way to know that the second one is on the way. In that case I see no good way to avoid a second recalculation.

((Indeed, I might suggest deprecating the two specific signals. They are convenient if you fully understand what gets called when, but the large number of similar signals makes it difficult to figure out that order.))